geostore/check_stac_metadata/utils.py

```diff
@@ -147,7 +151,17 @@ def process_assets(self) -> None:
 
     def validate(self, url: str) -> None:  # pylint: disable=too-complex
         self.traversed_urls.append(url)
-        object_json = self.get_object(url)
+        s3_response = self.get_object(url)
+        try:
+            object_json: JsonObject = load(
+                s3_response.response,
+                object_pairs_hook=self.duplicate_object_names_report_builder(url),
```

Comment:
Should this maybe be a separate PR? I basically forgot to include this functionality in at least one place (this one) where we parse metadata.